File: Swift/LeetCode/LeetCode1665.swift
// https://leetcode.com/problems/minimum-initial-energy-to-finish-tasks

class Solution {
    func minimumEffort(_ tasks: [[Int]]) -> Int {
        var answer = 0
        var current = 0
        let sortedTasks = tasks.sorted { return $0[1] - $0[0] > $1[1] - $1[0] }
        for task in sortedTasks {
            let (actual, minimum) = (task[0], task[1])
            if current < minimum {
                answer += minimum - current
                current = minimum
            }
            current -= actual
        }
        return answer
    }
}

File: Python/LeetCode/LeetCode1665.py
# https://leetcode.com/problems/minimum-initial-energy-to-finish-tasks

class Solution:
    def minimumEffort(self, tasks: List[List[int]]) -> int:
        tasks.sort(key = lambda x: x[1] - x[0], reverse = True)
        answer = 0
        current = 0
        for actual, minimum in tasks:
            if current < minimum:
                answer += minimum - current
                current = minimum
            current -= actual
        return answer
